### DISCUSSION TITLE
DependentServiceUnavailable

### DISCUSSION THREAD

**Discussion by EmilGH** [milo526/homebridge-tuya-web#384] [Q&A]
Hello,

I couldn't find this error message in the discussion thread… 

I'm trying to add a Tuya heater from Atomi Smart to Homebridge using this fabulous Tuya Web plug-in.  When I add the device and set to 'Thermostat', I get this error when adjusting temperature:

> Invalid payload in response: {"payload":{},"header":{"payloadVersion":1,"code":"DependentServiceUnavailable"}}

Conveniently, if you set the device to 'Switch', it can power on and off… but adjusting the temperature would be a lovely feature to have.

Any suggestions?

**Comment by milo526** [comment]
Hmmm interesting error indeed.
Could you run the script that is linked in the readme and show the output?
I'm afraid this device is improperly exposed by tuya as a switch instead of a thermostat and therefor these services are unavailable
